### PR TITLE
Refactor & Resolve git ref

### LINKED ISSUE #3: Lookup the head sha for the github repositories and generate stable resource urls
Instead of requesting the raw url, we should use https://github.com/adobe/helix-resolve-git-ref to resolve the head sha of the respective ref and pass this ref to the pipeline (or helix-static).

### PR DESCRIPTION
**Pull request by trieloff**
This will add one sequential request prior to retrieving content (or static content) from raw when the ref is a branch name and not a sha. This request resolves the sha for the branch name, allowing you to circumvent some of the built-in caching in raw, making sure to always retrieve the latest content

Fixes #3 

Also fixes refactoring issues: #10 #11

### CODE REVIEW

File: src/fetchers.js [hunk 1]
```diff
@@ -40,6 +41,10 @@ function errorPageResolver(res) {
   return defaultResolver(res);
 }
 
+function staticaction(contentOpts) {
```
Comment (tripodsan) [inline src/fetchers.js]: would be good to add some jsdoc and explain the `contentOpts` structure.

File: src/fetchers.js [hunk 2]
```diff
@@ -122,24 +230,21 @@ function fetchers(params = {}) {
   const dirindex = (params['content.index'] || 'index.html,README.html').split(',');
   const infos = getPathInfos(params.path || '/', params.rootPath || '', dirindex);
 
-  const staticOpts = {
+  const staticOpts = resolveOpts({
```
Comment (tripodsan) [inline src/fetchers.js]: add check to avoid double fetching if content and static repo are the same.

File: src/fetchers.js [hunk 3]
```diff
@@ -113,6 +118,109 @@ function getPathInfos(urlPath, mount, indices) {
...
+    name: 'helix-services/resolve-git-ref@v1',
+    blocking: true,
+    result: true,
+    params: opts,
+  }).then(res => ({
+    // use the resolved ref
+    ref: res.body.sha,
+    ...opts,
+  })).catch(() => opts); // if the resolver fails, just use the unresolved ref
+  return Promise.resolve(opts);
```
Comment (tripodsan) [inline src/fetchers.js]: I don't see how the result of the lookup is ever used...

File: src/fetchers.js [hunk 4]
```diff
@@ -122,24 +230,21 @@ function fetchers(params = {}) {
...
     owner: params['static.owner'],
     repo: params['static.repo'],
     ref: params['static.ref'],
     esi: params['static.esi'],
     root: params['static.root'],
-  };
+  });
 
-  const contentOpts = {
+  const contentOpts = resolveOpts({
```
Comment (tripodsan) [inline src/fetchers.js]: isn't `contentOpts` a promise now? if so, rename it something more descriptive.

File: src/fetchers.js [hunk 5]
```diff
@@ -113,6 +118,109 @@ function getPathInfos(urlPath, mount, indices) {
...
+      entry: info.path,
+      esi: false,
+      plain: true,
+      root: params['content.root'],
+      ...contentOpts,
+    },
+  })));
+}
+
+function resolveOpts(opts) {
```
[the remark this answers is not in the document]
Comment (trieloff) [reply src/fetchers.js]: Added the JSDoc. I know you prefer `async` over explicit Promises, my preferences are the opposite.

File: src/fetchers.js (hunk 2 above)
Comment (trieloff) [reply src/fetchers.js]: Done in 
0a21d29

**Pull request merged**